File: year2/mpp/c#/MotorcycleContest/MotorcycleContest/Domain/Validators/TeamValidator.cs
using MotorcycleContest.Domain.Entities;

namespace MotorcycleContest.Domain.Validators
{
    public class TeamValidator : IValidator<Team>
    {
        public void Validate(Team team)
        {
            if (team.Name.Equals(null))
            {
                throw new ValidationException("null fields");
            }
            if (team.Name.Equals(""))
            {
                throw new ValidationException("team is empty");
            }
        }
    }
}

File: year2/mpp/java/MotorcycleContest/src/main/java/domain/validators/TeamValidator.java
package domain.validators;

import domain.ValidationException;
import domain.entities.Team;
import org.springframework.stereotype.Component;

@Component
public class TeamValidator implements Validator<Team> {
    @Override
    public void validate(Team team) throws ValidationException {
        if (team.getName() == null) {
            throw new ValidationException("All team fields must be not null");
        }
        if (team.getName().equals("")) {
            throw new ValidationException("All team fields must be not empty");
        }
    }
}
